sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOIT.java

```diff
@@ -0,0 +1,190 @@
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.postgresql.ds.PGSimpleDataSource;
+
+
+/**
+ * A test of JdbcIO on an independent Postgres instance.
+ *
+ * <p>This test requires a running instance of Postgres, and the test dataset must exist in the
+ * databaase. `JdbcTestDataSet` will create the read table.
+ *
+ * <p>You can run just this test by doing the following:
+ * mvn test-compile compile failsafe:integration-test -D beamTestPipelineOptions='[
+ * "--postgresServerName=1.2.3.4",
+ * "--postgresUsername=postgres",
+ * "--postgresDatabaseName=myfancydb",
+ * "--postgresPassword=yourpassword",
+ * "--postgresSsl=false"
+ * ]' -DskipITs=false -Dit.test=org.apache.beam.sdk.io.jdbc.JdbcIOIT -DfailIfNoTests=false
+ */
+@RunWith(JUnit4.class)
+public class JdbcIOIT {
+  public PGSimpleDataSource getDataSource() throws SQLException {
+    PipelineOptionsFactory.register(PostgresTestOptions.class);
+    PostgresTestOptions options = TestPipeline.testingPipelineOptions()
+        .as(PostgresTestOptions.class);
+
+    return JdbcTestDataSet.getDataSource(options);
+  }
+  private static class CreateKVOfNameAndId implements JdbcIO.RowMapper<KV<String, Integer>> {
+    @Override
+    public KV<String, Integer> mapRow(ResultSet resultSet) throws Exception {
+      KV<String, Integer> kv =
+          KV.of(resultSet.getString("name"), resultSet.getInt("id"));
+      return kv;
+    }
+  }
+
+  private static class AssertCountFn implements
+      SerializableFunction<Iterable<KV<String, Long>>, Void> {
+    public AssertCountFn(long expectedCount) {
+     this.expectedCount = expectedCount;
+    }
+
+    private long expectedCount;
+
+    @Override
+    public Void apply(Iterable<KV<String, Long>> input) {
+      for (KV<String, Long> element : input) {
+        assertEquals(
+            element.getKey(), expectedCount, element.getValue().longValue());
+      }
+      return null;
+    }
+  }
+
+  private static class PutKeyInColumnOnePutValueInColumnTwo
+      implements JdbcIO.PreparedStatementSetter<KV<Integer, String>> {
+    @Override
+    public void setParameters(KV<Integer, String> element, PreparedStatement statement)
+                    throws SQLException {
+      statement.setInt(1, element.getKey());
+      statement.setString(2, element.getValue());
+    }
+  }
+
+  /**
+   * Does a test read of a few rows from a postgres database.
+   *
+   * <p>Note that IT read tests must not do any data table manipulation (setup/clean up.)
```

Comment:
The above guide I linked to is going to be public for everyone, and will be the standard practice for writing ITs. I don't expect every IT author to link to the reason, and I don't see how it adds value for me to - we don't expect PTransform authors to link to the PTransform style guide for everything they do. There are many things I'm doing in this that are based on the testing doc.